django/contrib/auth/models.py

```diff
@@ -142,8 +143,12 @@ def _create_user(self, username, email, password, **extra_fields):
         if not username:
             raise ValueError('The given username must be set')
         email = self.normalize_email(email)
-        username = self.model.normalize_username(username)
-        user = self.model(username=username, email=email, **extra_fields)
+        # Lookup the real model class from the global app registry so this
+        # manager method can be used in migrations. This is fine because
+        # managers are by definition working on the real model.
```

Comment:
> This is fine because managers are by definition working on the real model.

I thought adding the `use_in_migrations` flag to managers was done specially to allow working on models created in migrations?

This is going to break any migration that uses the return value of `create_user` as it will no longer be an instance of the user model class at the particular state the operations is run at. e.g.

```python
def forwards(apps, schema_editor):
    UserModel = apps.get_model('auth', 'User')
    user = UserModel.objects.create_user('Ms X', password='secure')
    assert isinstance(user, UserModel)
```

This is really problematic for migrations that use the returned user instance to assign foreign key values as they'll fail with `TypeError`.